transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java

```diff
@@ -383,6 +389,23 @@ private static InputStream maybeUnGzip(InputStream input, String contentEncoding
     return input;
   }
 
+  private String findPseudoId(BackendRequest request) {
+    Iterator<EventInternal> events = request.getEvents().iterator();
+
+    if(!events.hasNext()) return null;
+
+    String pseudoId = events.next().getPseudonymousId();
+
+    for(EventInternal event : request.getEvents()) {
```

Comment:
I think that assign the wrong cookie to an event is worst that assign none at all. If one of the events doesn't match, we should drop the cookie and return null.

Reply:
SGTM